--- libgnucash/engine/test/test-engine.c
+++ libgnucash/engine/test/test-engine.c
@@ -28,7 +28,6 @@
 extern void test_suite_budget();
 extern void test_suite_gncEntry();
 extern void test_suite_gncInvoice();
-extern void test_suite_query();
 extern void test_suite_transaction();
 extern void test_suite_policy();
 extern void test_suite_split();
@@ -48,7 +47,6 @@
     test_suite_account();
     test_suite_budget();
     test_suite_gncEntry();
-    test_suite_query();
     test_suite_gncInvoice();
     test_suite_transaction();
     test_suite_policy();
