diff -urN old.guile-1.8.7/configure.in new/guile-1.8.7/configure.in
--- old.guile-1.8.7/configure.in  2009-07-05 14:40:06.000000000 -0400
+++ new.guile-1.8.7/configure.in  2009-08-26 15:33:43.996500000 -0400
@@ -25,7 +25,7 @@
 
 ]])
 
-AC_PREREQ(2.61)
+AC_PREREQ(2.63)
 
 dnl  `patsubst' here deletes the newline which "echo" prints.  We can't use
 dnl  "echo -n" since -n is not portable (see autoconf manual "Limitations of
@@ -41,7 +41,7 @@
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_SRCDIR(GUILE-VERSION)
 
-AM_INIT_AUTOMAKE([gnu no-define check-news -Wall -Wno-override])
+AM_INIT_AUTOMAKE([1.11 silent-rules gnu no-define check-news -Wall -Wno-override])
 
 AC_COPYRIGHT(GUILE_CONFIGURE_COPYRIGHT)
 AC_CONFIG_SRCDIR([GUILE-VERSION])
@@ -68,8 +68,6 @@
 
 AB_INIT
 
-AC_LIBTOOL_WIN32_DLL
-
 AC_PROG_INSTALL
 AC_PROG_CC
 AC_PROG_CPP
@@ -82,8 +80,7 @@
 # for per-target cflags in the libguile subdir
 AM_PROG_CC_C_O
 
-AC_LIBTOOL_DLOPEN
-AC_PROG_LIBTOOL
+LT_INIT([win32-dll dlopen])
 AC_CHECK_LIB([ltdl], [lt_dlinit], ,
   [AC_MSG_ERROR([libltdl not found.  See README.])])
 AC_CHECK_HEADER([ltdl.h], [],
diff -urN old.guile-1.8.7/guile-readline/configure.in new/guile-1.8.7/guile-readline/configure.in
--- old.guile-1.8.7/guile-readline/configure.in 2009-07-03 18:19:22.000000000 -0400
+++ new.guile-1.8.7/guile-readline/configure.in 2009-08-26 15:35:08.398500000 -0400
@@ -1,4 +1,4 @@
-AC_PREREQ(2.50)
+AC_PREREQ(2.63)
 
 dnl Don't use "echo -n", which is not portable (e.g., not available on
 dnl MacOS X).  Instead, use `patsubst' to remove the newline.
@@ -10,15 +10,14 @@
 AC_CONFIG_AUX_DIR([.])
 AC_CONFIG_SRCDIR(readline.c)
 AM_CONFIG_HEADER([guile-readline-config.h])
-AM_INIT_AUTOMAKE([foreign no-define])
+AM_INIT_AUTOMAKE([1.11 silent-rules foreign no-define])
 
 . $srcdir/../GUILE-VERSION
 
 AC_PROG_INSTALL
 AC_PROG_CC
 AM_PROG_CC_STDC
-AC_LIBTOOL_WIN32_DLL
-AC_PROG_LIBTOOL
+LT_INIT([win32-dll])
 
 dnl
 dnl Check for Winsock and other functionality on Win32 (*not* CygWin)
--- old.guile-1.8.7/pre-inst-guile.in	2010-05-30 09:38:50 +0000
+++ new.guile-1.8.7/pre-inst-guile.in	2010-05-30 09:39:30 +0000
@@ -87,7 +87,7 @@
 export DYLD_LIBRARY_PATH
 
 # set GUILE (clobber)
-GUILE=${top_builddir}/libguile/guile
+GUILE=${top_builddir}/libguile/.libs/guile
 export GUILE
 
 # do it
===================================================================
--- sid.orig/libguile/threads.h
+++ sid/libguile/threads.h
@@ -26,7 +26,6 @@
 #include "libguile/procs.h"
 #include "libguile/throw.h"
 #include "libguile/root.h"
-#include "libguile/iselect.h"
 #include "libguile/dynwind.h"
 #include "libguile/continuations.h"
 
