--- libxslt/Makefile.in.orig	2007-08-23 11:22:57 -0400
+++ libxslt/Makefile.in	2011-04-05 17:11:09 -0400
@@ -293,8 +293,8 @@
 	xsltwin32config.h.in		\
 	libxslt.h
 
-libxslt_la_LIBADD = $(EXTRA_LIBS)
-libxslt_la_LDFLAGS = -version-info @LIBXSLT_VERSION_INFO@
+libxslt_la_LIBADD = $(EXTRA_LIBS) $(WIN32_EXTRA_LIBADD)
+libxslt_la_LDFLAGS = -version-info @LIBXSLT_VERSION_INFO@ $(WIN32_EXTRA_LDFLAGS)
 man_MANS = libxslt.3
 EXTRA_DIST = $(man_MANS) trio.h triodef.h
 all: all-am
--- libexslt/Makefile.in.orig	2007-08-23 11:22:57 -0400
+++ libexslt/Makefile.in	2011-04-05 17:11:44 -0400
@@ -269,8 +269,8 @@
 	libexslt.h			\
 	dynamic.c
 
-libexslt_la_LIBADD = $(top_builddir)/libxslt/libxslt.la $(EXTRA_LIBS)
-libexslt_la_LDFLAGS = -version-info @LIBEXSLT_VERSION_INFO@ $(LIBGCRYPT_LIBS)
+libexslt_la_LIBADD = $(top_builddir)/libxslt/libxslt.la $(EXTRA_LIBS) $(WIN32_EXTRA_LIBADD)
+libexslt_la_LDFLAGS = -version-info @LIBEXSLT_VERSION_INFO@ $(LIBGCRYPT_LIBS) $(WIN32_EXTRA_LDFLAGS)
 man_MANS = libexslt.3
 EXTRA_DIST = $(man_MANS)
 all: all-am
