--- lib/Makefile.am~	Thu Dec 15 13:13:16 2005
+++ lib/Makefile.am	Tue Feb  6 10:57:58 2007
@@ -76,7 +76,9 @@
 	splibpch.h token.h app_inst.cxx arc_inst.cxx entmgr_inst.cxx \
 	parser_inst.cxx xentmgr_inst.cxx SubstTable.cxx \
 	UTF16CodingSystem.cxx Fixed4CodingSystem.cxx \
-	memcmp.c memmove.c strerror.c
+	memcmp.c memmove.c strerror.c WinInetStorage.cxx \
+	WinInetStorageMessages.h WinInetStorageMessages.msg \
+	WinInetStorageMessages.rc WinApp.cxx Win32CodingSystem.cxx
 
 INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/generic
 
