diff -ur webkit-1.2.7-orig/WebCore/dom/XMLTokenizerLibxml2.cpp webkit-1.2.7/WebCore/dom/XMLTokenizerLibxml2.cpp
--- webkit-1.2.7-orig/WebCore/dom/XMLTokenizerLibxml2.cpp	2010-09-10 15:20:33.000000000 +0200
+++ webkit-1.2.7/WebCore/dom/XMLTokenizerLibxml2.cpp	2011-03-12 23:13:39.521370554 +0100
@@ -23,6 +23,9 @@
  * Boston, MA 02110-1301, USA.
  */
 
+#define _GNU_SOURCE
+#include <cstdio>
+
 #include "config.h"
 #include "XMLTokenizer.h"
 
@@ -908,7 +911,7 @@
     if (m_parserStopped)
         return;
 
-#if COMPILER(MSVC) || COMPILER(RVCT)
+#if 1
     char m[1024];
     vsnprintf(m, sizeof(m) - 1, message, args);
 #else
