--- configure	Tue Oct 05 06:02:37 2010
+++ configure	Tue Jan 11 07:28:51 2011
@@ -1077,6 +1077,7 @@
 enable_icon_database
 enable_datalist
 enable_ruby
+enable_sandbox
 enable_video
 enable_xhtmlmp
 enable_xpath
@@ -19020,9 +19021,9 @@
 # check whether to enable HTML5 sandbox iframe support
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether to enable HTML5 sandboxed iframe support" >&5
 $as_echo_n "checking whether to enable HTML5 sandboxed iframe support... " >&6; }
-# Check whether --enable-ruby was given.
-if test "${enable_ruby+set}" = set; then :
-  enableval=$enable_ruby;
+# Check whether --enable-sandbox was given.
+if test "${enable_sandbox+set}" = set; then :
+  enableval=$enable_sandbox;
 else
   enable_sandbox="yes"
 fi
