--- WebCore/bindings/scripts/IDLParser.pm	Thu Jan 15 16:14:12 1970
+++ WebCore/bindings/scripts/IDLParser.pm	Thu Jan 15 16:14:12 1970
@@ -69,7 +69,7 @@
         if (($Config::Config{'osname'}) =~ /solaris/i) {
             $gccLocation = "/usr/sfw/bin/gcc";
         } else {
-            $gccLocation = "/usr/bin/gcc";
+            $gccLocation = "gcc";
         }
         $preprocessor = $gccLocation . " -E -P -x c++";
     }
--- WebCore/css/make-css-file-arrays.pl	Thu Jan 15 16:14:12 1970
+++ WebCore/css/make-css-file-arrays.pl	Thu Jan 15 16:14:12 1970
@@ -33,7 +33,7 @@
     if (($Config::Config{'osname'}) =~ /solaris/i) {
         $gccLocation = "/usr/sfw/bin/gcc";
     } else {
-        $gccLocation = "/usr/bin/gcc";
+        $gccLocation = "gcc";
     }
     $preprocessor = $gccLocation . " -E -P -x c++";
 }
--- WebCore/dom/make_names.pl	Thu Jan 15 16:14:12 1970
+++ WebCore/dom/make_names.pl	Thu Jan 15 16:14:12 1970
@@ -52,7 +52,7 @@
 if (($Config::Config{'osname'}) =~ /solaris/i) {
     $gccLocation = "/usr/sfw/bin/gcc";
 } else {
-    $gccLocation = "/usr/bin/gcc";
+    $gccLocation = "gcc";
 }
 my $preprocessor = $gccLocation . " -E -P -x c++";
 
