--- GNUmakefile	Mon Jan 17 09:32:30 2011
+++ GNUmakefile	Mon Jan 17 09:44:19 2011
@@ -6411,3 +6411,4 @@
 STRIP = strip
-SYMBOL_VISIBILITY = -fvisibility=hidden
-SYMBOL_VISIBILITY_INLINES = -fvisibility-inlines-hidden
+SYMBOL_VISIBILITY = 
+SYMBOL_VISIBILITY_INLINES = 
+PTHREAD_LIBS = -lpthread
@@ -8967,7 +8967,7 @@
 # It breaks the build on other platforms, so we use it conditionally
 no_undefined = -no-undefined
 #version_script = -Wl,--version-script,$(srcdir)/autotools/symbols.filter
-version_script = -export-symbols-regex "^(webkit_|JS).*"
+version_script = -export-symbols-regex "^(webkit_|k?JS).*"
 
 # Shared libraries
 lib_LTLIBRARIES = \
@@ -8987,7 +8987,7 @@
 libJavaScriptCore_la_LIBADD = \
 	$(UNICODE_LIBS) \
 	$(GLIB_LIBS) \
-	-lpthread
+	$(PTHREAD_LIBS)
 
 libJavaScriptCore_la_CXXFLAGS = \
 	$(global_cxxflags) \
@@ -9042,7 +9042,6 @@
 	$(no_undefined)
 
 libwebkit_1_0_la_LIBADD = \
-	-lpthread \
 	libJavaScriptCore.la \
 	libWebCoreJS.la \
 	$(webcore_ldflags) \
@@ -9064,7 +9063,8 @@
 	$(PNG_LIBS) \
 	$(SQLITE3_LIBS) \
 	$(UNICODE_LIBS) \
-	$(XT_LIBS)
+	$(PTHREAD_LIBS) \
+	$(XT_LIBS) -lwinmm -lshlwapi -lole32
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = WebKit/gtk/webkit-1.0.pc
@@ -9104,7 +9104,7 @@
 Programs_minidom_LDADD = \
 	 libJavaScriptCore.la \
 	 -lm \
-	 -lstdc++
+	 -lstdc++ -lwinmm
 
 Programs_minidom_LDFLAGS = \
 	-no-install \
@@ -9127,7 +9127,7 @@
 	$(UNICODE_CFLAGS)
 
 Programs_jsc_LDADD = \
-	libJavaScriptCore.la
+	libJavaScriptCore.la -lwinmm
 
 WebCore := $(srcdir)/WebCore
 
@@ -9263,7 +9263,7 @@
 Programs_GtkLauncher_LDADD = \
 	libwebkit-1.0.la \
 	$(GTK_LIBS) \
-	$(GLIB_LIBS)
+	$(GLIB_LIBS) -lwinmm
 
 Programs_GtkLauncher_LDFLAGS = \
 	-no-fast-install \
@@ -9326,7 +9326,7 @@
 	$(GTK_LIBS) \
 	$(GLIB_LIBS) \
 	$(LIBSOUP_LIBS) \
-	$(FREETYPE_LIBS)
+	$(FREETYPE_LIBS) -lwinmm
 
 Programs_DumpRenderTree_LDFLAGS = \
 	-no-fast-install \
