diff -ur -x '*.lo' -x '*.o' webkit-1.1.90-orig/WebCore/platform/network/ResourceResponseBase.h webkit-1.1.90/WebCore/platform/network/ResourceResponseBase.h
--- webkit-1.1.90-orig/WebCore/platform/network/ResourceResponseBase.h	2010-02-22 17:36:04.000000000 +0100
+++ webkit-1.1.90/WebCore/platform/network/ResourceResponseBase.h	2011-03-12 22:11:06.720211214 +0100
@@ -31,6 +31,7 @@
 #include "KURL.h"
 
 #include <wtf/PassOwnPtr.h>
+#include <ctime>
 
 namespace WebCore {
 
diff -ur webkit-1.2.7-orig/WebCore/page/Page.h webkit-1.2.7/WebCore/page/Page.h
--- webkit-1.2.7-orig/WebCore/page/Page.h	2010-09-10 15:20:33.000000000 +0200
+++ webkit-1.2.7/WebCore/page/Page.h	2011-03-12 23:25:35.670211006 +0100
@@ -26,6 +26,7 @@
 #include <wtf/Forward.h>
 #include <wtf/HashSet.h>
 #include <wtf/Noncopyable.h>
+#include <ctime>
 
 #if PLATFORM(MAC)
 #include "SchedulePair.h"
