diff -Nur aqbanking-3.7.1-old/configure.ac aqbanking-3.7.1/configure.ac
--- aqbanking-3.7.1-old/configure.ac	2008-08-16 13:36:42.000000000 +0200
+++ aqbanking-3.7.1/configure.ac	2008-10-05 18:56:36.925980908 +0200
@@ -476,7 +476,7 @@
 AC_PROG_INSTALL
 
 AC_LIBTOOL_WIN32_DLL
-AC_LIBTOOL_RC
+AC_CHECK_TOOL([RC], [windres], [no])
 AC_PROG_LIBTOOL
 AC_CHECK_PROG(USE_DOT,dot,YES,NO)
 AC_CHECK_PROG(SED,sed,sed)
@@ -511,7 +511,6 @@
 #
 # Checks for library functions.
 #
-AC_FUNC_MALLOC
 AC_FUNC_STRFTIME
 AC_CHECK_FUNCS([memmove memset strcasecmp strdup strerror snprintf])
 AC_CHECK_FUNCS([setlocale])
diff -Nur aqbanking-3.7.1-old/src/frontends/qbanking/bin/helpbrowser/Makefile.am aqbanking-3.7.1/src/frontends/qbanking/bin/helpbrowser/Makefile.am
--- aqbanking-3.7.1-old/src/frontends/qbanking/bin/helpbrowser/Makefile.am	2008-07-31 20:13:31.000000000 +0200
+++ aqbanking-3.7.1/src/frontends/qbanking/bin/helpbrowser/Makefile.am	2008-10-05 18:56:21.957982868 +0200
@@ -65,7 +65,7 @@
 
 # This section is for a qt4 environment
 qt4-port:
-	for A in $(libqbhelp_la_SOURCES) $(noinst_HEADERS); do \
+	for A in $(libqbhelp_la_SOURCES) $(noinst_HEADERS) qb_help.cpp qbhelpbrowser.cpp; do \
 	 $(YES) | $(QT3TO4) $$A; \
 	done
 #	rm $(BUILT_SOURCES)
