--- trunk/src/plugins/parsers/dtaus/dtaus-export.c	2011/01/11 19:59:25	2131
+++ trunk/src/plugins/parsers/dtaus/dtaus-export.c	2011/01/13 21:47:48	2132
@@ -531,7 +531,7 @@
       return -1;
     }
     AB_Value_AddValue(sumDEM, val);
-    snprintf(buffer, sizeof(buffer), "%011.0lf", AB_Value_GetValueAsDouble(val)*100.0);
+    snprintf(buffer, sizeof(buffer), "%011.0f", AB_Value_GetValueAsDouble(val)*100.0);
     AB_Value_free(val);
     if (AHB_DTAUS__AddNum(dst, 11, buffer)) {
       DBG_ERROR(AQBANKING_LOGDOMAIN, "Error writing to buffer");
@@ -587,7 +587,7 @@
       return -1;
     }
     AB_Value_AddValue(sumEUR, val);
-    snprintf(buffer, sizeof(buffer), "%011.0lf", AB_Value_GetValueAsDouble(val)*100.0);
+    snprintf(buffer, sizeof(buffer), "%011.0f", AB_Value_GetValueAsDouble(val)*100.0);
     AB_Value_free(val);
     if (AHB_DTAUS__AddNum(dst, 11, buffer)) {
       DBG_ERROR(AQBANKING_LOGDOMAIN, "Error writing to buffer");
@@ -768,7 +768,7 @@
   }
 
   /* field 5: sum of DEM values */
-  snprintf(buffer, sizeof(buffer), "%013.0lf",
+  snprintf(buffer, sizeof(buffer), "%013.0f",
 	   AB_Value_GetValueAsDouble(sumDEM)*100.0);
   if (AHB_DTAUS__AddNum(dst, 13, buffer)) {
     DBG_ERROR(AQBANKING_LOGDOMAIN, "Error writing to buffer");
@@ -776,7 +776,7 @@
   }
 
   /* field 6: sum of peer account ids */
-  snprintf(buffer, sizeof(buffer), "%017.0lf",
+  snprintf(buffer, sizeof(buffer), "%017.0f",
 	   AB_Value_GetValueAsDouble(sumAccountIds));
   if (AHB_DTAUS__AddNum(dst, 17, buffer)) {
     DBG_ERROR(AQBANKING_LOGDOMAIN, "Error writing to buffer");
@@ -784,7 +784,7 @@
   }
 
   /* field 7: sum of peer bank codes */
-  snprintf(buffer, sizeof(buffer), "%017.0lf",
+  snprintf(buffer, sizeof(buffer), "%017.0f",
 	   AB_Value_GetValueAsDouble(sumBankCodes));
   if (AHB_DTAUS__AddNum(dst, 17, buffer)) {
     DBG_ERROR(AQBANKING_LOGDOMAIN, "Error writing to buffer");
@@ -792,7 +792,7 @@
   }
 
   /* field 8: sum of EUR values */
-  snprintf(buffer, sizeof(buffer), "%013.0lf",
+  snprintf(buffer, sizeof(buffer), "%013.0f",
 	   AB_Value_GetValueAsDouble(sumEUR)*100.0);
   if (AHB_DTAUS__AddNum(dst, 13, buffer)) {
     DBG_ERROR(AQBANKING_LOGDOMAIN, "Error writing to buffer");
