diff -Nur goffice-0.3.0-old/configure.in goffice-0.3.0/configure.in
--- goffice-0.3.0-old/configure.in	Mon Apr 17 19:15:44 2006
+++ goffice-0.3.0/configure.in	Wed Oct 11 16:56:52 2006
@@ -93,6 +93,7 @@
 	libxml-2.0		>= 2.4.12
 	pango			>= 1.8.1
 	pangoft2		>= 1.8.1
+	freetype2		>= 2.1.0
 	libart-2.0		>= 2.3.11
 "
 goffice_cairo_reqs="
@@ -225,6 +226,8 @@
 AC_MSG_RESULT($with_win32)
 AM_CONDITIONAL(WITH_WIN32, test $with_win32 = yes)
 AM_CONDITIONAL(HAVE_LIBEXE, test x$LIBEXE = xyes)
+
+AC_CHECK_HEADERS([pwd.h grp.h])
 
 dnl *****************************
 dnl FIXME: perhaps declare with AC_ARG_VAR?
diff -Nur goffice-0.3.0-old/goffice/Makefile.am goffice-0.3.0/goffice/Makefile.am
--- goffice-0.3.0-old/goffice/Makefile.am	Mon Apr 17 20:53:01 2006
+++ goffice-0.3.0/goffice/Makefile.am	Wed Oct 11 16:56:52 2006
@@ -30,7 +30,7 @@
 if WITH_WIN32
 libgoffice_0_la_DEPENDENCIES = goffice.def
 libgoffice_0_la_LDFLAGS += -no-undefined -export-symbols goffice.def
-libgoffice_0_la_LIBADD += -luuid -lhtmlhelp -lurlmon
+libgoffice_0_la_LIBADD += -luuid
 endif
 
 BUILT_SOURCES =	goffice-paths.h
@@ -66,14 +66,14 @@
 include $(top_srcdir)/goffice.mk
 
 if WITH_WIN32
-LIB_PUBLIC_HDRS = $(libgoffice_1_la_HEADERS)
+LIB_PUBLIC_HDRS = $(libgoffice_0_la_HEADERS)
 
 goffice.def: local.def app/local.def \
 	cut-n-paste/foocanvas/local.def \
 	data/local.def drawing/local.def graph/local.def \
 	gtk/local.def ms-compat/local.def utils/local.def
 	echo EXPORTS > $@ && \
-	cat $^ | sort >> $@
+	cat $^ | sort | sed '/^go_doc_mark_not_modified$$/d;/^go_plugin_init$$/d;/^go_plugin_shutdown$$/d' >> $@
 
 if HAVE_LIBEXE
 goffice_mslibdir = $(DESTDIR)$(libdir)
diff -Nur goffice-0.3.0-old/goffice/utils/go-file.c goffice-0.3.0/goffice/utils/go-file.c
--- goffice-0.3.0-old/goffice/utils/go-file.c	Wed Mar 29 15:41:37 2006
+++ goffice-0.3.0/goffice/utils/go-file.c	Wed Oct 11 16:56:52 2006
@@ -44,8 +44,12 @@
 #include <string.h>
 #include <stdlib.h>
 #include <unistd.h>
-#include <pwd.h>
-#include <grp.h>
+#ifdef HAVE_PWD_H
+# include <pwd.h>
+#endif
+#ifdef HAVE_GRP_H
+# include <grp.h>
+#endif
 #include <time.h>
 /* ------------------------------------------------------------------------- */
 
@@ -675,11 +679,13 @@
 {
 	gboolean error = FALSE;
 	guint uid = 0;
+	gboolean islocal = FALSE;
+#ifdef HAVE_PWD_H
 	struct passwd *password_info;
 	const char *name;
 	gsize namelen;
 	char *nameutf8;
-	gboolean islocal = FALSE;
+#endif
 
 #ifdef GOFFICE_WITH_GNOME
 	GnomeVFSFileInfo *file_info;
@@ -718,6 +724,7 @@
 		return g_strdup (_("remote user"));
 	}
 
+#ifdef HAVE_PWD_H
 	password_info = getpwuid (uid);
 
 	if (password_info == NULL)
@@ -739,6 +746,9 @@
 		nameutf8[--namelen] = 0;
 
 	return nameutf8;
+#else
+	return NULL;
+#endif
 }
 
 /*
@@ -750,10 +760,12 @@
 {
 	gboolean error = FALSE;
 	guint gid = 0;
-	struct group *group_info;
 	gboolean islocal = FALSE;
+#ifdef HAVE_GRP_H
+	struct group *group_info;
 	const char *name;
 	char *nameutf8;
+#endif
 
 #ifdef GOFFICE_WITH_GNOME
 	GnomeVFSFileInfo *file_info;
@@ -793,6 +805,7 @@
 		return g_strdup (_("remote"));
 	}
 
+#ifdef HAVE_GRP_H
 	group_info = getgrgid (gid);
 
 	if (group_info == NULL)
@@ -802,6 +815,9 @@
 	(void) go_guess_encoding (name, strlen (name),
 				  NULL, &nameutf8);
 	return nameutf8;
+#else
+	return NULL;
+#endif
 }
 
 GOFilePermissions *
diff -Nur goffice-0.3.0-old/gtk-doc.m4 goffice-0.3.0/gtk-doc.m4
--- goffice-0.3.0-old/gtk-doc.m4	Thu Jan  1 00:00:00 1970
+++ goffice-0.3.0/gtk-doc.m4	Wed Oct 11 16:57:21 2006
@@ -0,0 +1,53 @@
+dnl -*- mode: autoconf -*-
+
+# serial 1
+
+dnl Usage:
+dnl   GTK_DOC_CHECK([minimum-gtk-doc-version])
+AC_DEFUN([GTK_DOC_CHECK],
+[
+  AC_BEFORE([AC_PROG_LIBTOOL],[$0])dnl setup libtool first
+  AC_BEFORE([AM_PROG_LIBTOOL],[$0])dnl setup libtool first
+  dnl for overriding the documentation installation directory
+  AC_ARG_WITH(html-dir,
+    AC_HELP_STRING([--with-html-dir=PATH], [path to installed docs]),,
+    [with_html_dir='${datadir}/gtk-doc/html'])
+  HTML_DIR="$with_html_dir"
+  AC_SUBST(HTML_DIR)
+
+  dnl enable/disable documentation building
+  AC_ARG_ENABLE(gtk-doc,
+    AC_HELP_STRING([--enable-gtk-doc],
+                   [use gtk-doc to build documentation [default=no]]),,
+    enable_gtk_doc=no)
+
+  have_gtk_doc=no
+  if test x$enable_gtk_doc = xyes; then
+    if test -z "$PKG_CONFIG"; then
+      AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
+    fi
+    if test "$PKG_CONFIG" != "no" && $PKG_CONFIG --exists gtk-doc; then
+      have_gtk_doc=yes
+    fi
+
+  dnl do we want to do a version check?
+ifelse([$1],[],,
+    [gtk_doc_min_version=$1
+    if test "$have_gtk_doc" = yes; then
+      AC_MSG_CHECKING([gtk-doc version >= $gtk_doc_min_version])
+      if $PKG_CONFIG --atleast-version $gtk_doc_min_version gtk-doc; then
+        AC_MSG_RESULT(yes)
+      else
+        AC_MSG_RESULT(no)
+        have_gtk_doc=no
+      fi
+    fi
+])
+    if test "$have_gtk_doc" != yes; then
+      enable_gtk_doc=no
+    fi
+  fi
+
+  AM_CONDITIONAL(ENABLE_GTK_DOC, test x$enable_gtk_doc = xyes)
+  AM_CONDITIONAL(GTK_DOC_USE_LIBTOOL, test -n "$LIBTOOL")
+])
