--- icu.orig/source/config/mh-mingw	2010-04-29 00:27:52.000000000 +0900
+++ icu/source/config/mh-mingw	2010-05-02 16:18:18.821561705 +0900
@@ -71,8 +71,8 @@
 
 # The #M# is used to delete lines for icu-config
 # Current full path directory.
-#CURR_FULL_DIR=$(shell pwd -W)#M# for MSYS
-CURR_FULL_DIR=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
+CURR_FULL_DIR=$(shell pwd)#M# for MSYS
+#CURR_FULL_DIR=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
 # Current full path directory for use in source code in a -D compiler option.
 #CURR_SRCCODE_FULL_DIR=$(subst /,\\\\,$(shell pwd -W))#M# for MSYS
 CURR_SRCCODE_FULL_DIR=$(subst \,/,$(shell cmd /c cd))#M# for Cygwin shell
