--- drivers/pgsql/dbd_pgsql.c	Sat Jan 26 12:39:50 2008
+++ drivers/pgsql/dbd_pgsql.c.new	Tue Jun  9 21:06:14 2009
@@ -520,11 +520,11 @@
   return db;
 }
 
-int dbd_geterror(dbi_conn_t *conn, int *errno, char **errstr) {
-	/* put error number into errno, error string into errstr
-	 * return 0 if error, 1 if errno filled, 2 if errstr filled, 3 if both errno and errstr filled */
+int dbd_geterror(dbi_conn_t *conn, int *err_no, char **errstr) {
+	/* put error number into err_no, error string into errstr
+	 * return 0 if error, 1 if err_no filled, 2 if errstr filled, 3 if both err_no and errstr filled */
 	
-	*errno = 0;
+	*err_no = 0;
 	*errstr = strdup(PQerrorMessage((PGconn *)conn->connection));
 	
 	return 2;
