--- lib/ofx_preproc.cpp~	Tue Jan  9 02:38:33 2007
+++ lib/ofx_preproc.cpp	Tue Feb  6 12:31:07 2007
@@ -20,6 +20,7 @@
 #include <iostream>
 #include <fstream>
 #include <stdlib.h>
+#include <io.h> // for mktemp() on win32/mingw
 #include <stdio.h>
 #include <string>
 #include "ParserEventGeneratorKit.h"
@@ -76,7 +77,7 @@
     
     input_file.open(p_filename);
     strncpy(tmp_filename,"/tmp/libofxtmpXXXXXX",50);
-    mkstemp(tmp_filename);
+    mktemp(tmp_filename);
     tmp_file.open(tmp_filename);
 
     message_out(DEBUG,"ofx_proc_file(): Creating temp file: "+string(tmp_filename));
@@ -217,7 +218,7 @@
   s_buffer=string(s, size);
 
   strncpy(tmp_filename,"/tmp/libofxtmpXXXXXX",50);
-  mkstemp(tmp_filename);
+  mktemp(tmp_filename);
   tmp_file.open(tmp_filename);
 
   message_out(DEBUG,"ofx_proc_file(): Creating temp file: "+string(tmp_filename));
